### PR TITLE
Fix add spark dependency to expose jars via the class file handler

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Review by albertshau — APPROVED** [review]
lgtm

File: cdap-spark-core-base/src/main/java/co/cask/cdap/app/runtime/spark/dynamic/SparkClassFileHandler.java [hunk 1]
```diff
@@ -18,60 +18,115 @@
...
 
-    for (File dir : classDirs) {
-      File classFile = new File(dir, requestURI);
-      if (classFile.isFile()) {
-        responder.sendByteArray(HttpResponseStatus.OK, Files.readAllBytes(classFile.toPath()),
+    Lock readLock = this.lock.readLock();
+    readLock.lock();
+    try {
+      // Use the classloader to find the class file
+      URL resource = jarClassFinder.getResource(requestURI);
```
Comment (YanAtCask) [inline cdap-spark-core-base/src/main/java/co/cask/cdap/app/runtime/spark/dynamic/SparkClassFileHandler.java:120]: Is it good enough to apply readLock to getResource()?

Comment (chtyim) [reply cdap-spark-core-base/src/main/java/co/cask/cdap/app/runtime/spark/dynamic/SparkClassFileHandler.java:120]: I think so. Why?

Comment (YanAtCask) [reply cdap-spark-core-base/src/main/java/co/cask/cdap/app/runtime/spark/dynamic/SparkClassFileHandler.java:120]: I struggled a little bit between "return" and "finally".

Comment (chtyim) [reply cdap-spark-core-base/src/main/java/co/cask/cdap/app/runtime/spark/dynamic/SparkClassFileHandler.java:120]: `finally` always get executed regardless of `return` or `exception`.

Comment (YanAtCask) [reply cdap-spark-core-base/src/main/java/co/cask/cdap/app/runtime/spark/dynamic/SparkClassFileHandler.java:120]: Thank you!

**Comment by chtyim** [comment]
Buikld passed https://builds.cask.co/browse/CDAP-RUT1071

**Comment by chtyim** [comment]
Squashed commits.

**Pull request merged**